docs/modules/qbi-helper/interest-template.md

```diff
@@ -0,0 +1,141 @@
+
+You initialise an empty [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) by calling `InterestTemplate.Create()`. This is useful when defining a set of interest queries for the first time, for example when defining Entity Templates.
+
+```csharp
+var basicInterestTemplate = InterestTemplate.Create();
+```
+
+You can also construct this from an existing [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) or set of interest queries. This creates a deep copy of the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template), to allow you to modify the queries without affecting the original set.
+
+```csharp
+var advancedInterestTemplate = InterestTemplate.Create(basicInterestTemplate);
+```
+
+## Modify
+
+When modifying an [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template), you must specify which particular component's queries you are modifying. You can either:
+
+* provide the component as a type argument
+* pass the component ID as the first argument
+
+For example, to add a query you might do one of the following:
+
+```csharp
+// Component identified by type argument
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Component identified by ID given as first argument
+InterestTemplate.Create()
+    .AddQueries(Position.ComponentId, query1, query2);
+```
+
+### Add
+
+When adding queries, you can either provide the queries as parameters or an enumerable set:
+
+```csharp
+// Parameters
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Enumerable
+var queryList = new List() { query1, query2 };
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(queryList);
+```
+
+### Replace
+
+By doing this, you replace all the existing queries for an authoritative component with the new queries you pass in. In a similar way to adding queries, you must provide the queries as parameters or an enumerable set.
+
+```csharp
+// Parameters
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(query1);
+
+// Enumerable
+var queryList = new List() { query1 };
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(queryList);
+```
+
+At least one query must be provided, otherwise the existing queries for a given authoritative component will not be removed.
+
+### Clear
+
+You can either clear queries for a given authoritative component or clear _all_ queries in the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
+
+```csharp
+// Removes all queries added for the Position component
+basicInterestTemplate.ClearQueries<Position.Component>();
+
+// Removes all queries
+basicInterestTemplate.ClearAllQueries();
+```
+
+After adding, removing or modifying a set of queries, there are two ways to get `Interest` out of the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
```

Comment:
This acts as a segue into the next section. But its feels like its part of `Clear`.

Suggest breaking this into two parts: `Modify` and `Apply Interest` or similar.

Then the segue can go under the `Apply Interest` header before you dive into `Snapshot` or `ComponentInterest`

Reply:
yup this is how I had it but I seem to have deleted the heading 😬 (adding it back)